### ISSUE TITLE
Form Key Error with Pager on List Views 

### ISSUE THREAD

**Issue by inavdb** [hyva-themes/magento2-hyva-admin#56]
_Error occures in the integer_net Slider module (see also internal ticket 17261)_

When in a list the number of elements exceeds the standard qty er page (here 20), you cannot edit an element when
1. You navigate to page 2 and try to open an element.
2. Switching the rows per page to e.g. 50, trying to edit en element any element

Instead, there's a form key error and you end up on the Dashboard.
Editing one of the first 20 elements, when the elements per page = 20 works though.

**Comment by Vinai** [comment]
Note to self: it seems the form key changes when an Ajax grid update id done.

**Comment by Vinai** [comment]
I can see the action URL is configured as ` <action id="edit" label="Edit" url="*/*/edit"` in the grid XML.
This configuration doesn't work for Ajax updates of the grid, since they happen through a generic Ajax pager route.
Applying the following patch to the grid config should fix the issue:

```diff
diff --git a/view/adminhtml/hyva-grid/hyva-slider-grid.xml b/view/adminhtml/hyva-grid/hyva-slider-grid.xml
index 3b5f08e..b9d8a9f 100644
--- a/view/adminhtml/hyva-grid/hyva-slider-grid.xml
+++ b/view/adminhtml/hyva-grid/hyva-slider-grid.xml
@@ -17,7 +17,7 @@
         </exclude>
     </columns>
     <actions idColumn="slider_id">
-        <action id="edit" label="Edit" url="*/*/edit" idParam="slider"/>
+        <action id="edit" label="Edit" url="integernet_hyvaslider/slider/edit" idParam="slider"/>
     </actions>
     <navigation>
         <buttons>
```

I'll leave the issue open for now until you can confirm this fixes the problem.

**Comment by Vinai** [comment]
I've patched the `UrlBuilder` during Ajax requests so it will use the original route to build new URLs.
Now it will work even without a config change.
This will be part of the 1.1.16 release.

**Comment by lbuchholz** [comment]
@Vinai I can confirm this working for next page navigation, but clicking previous page link won't update anything.

**Comment by Vinai** [comment]
Thanks for letting me know. I'll look into this, but it will be a while because vacation.

**Comment by Vinai** [comment]
@lbuchholz I can't reproduce the issue paging through sliders or the test grids.
Is there any additional info you can give?

**Comment by lbuchholz** [comment]
Not really... Use case is hyva_slider with 40+ sliders, page size 20. While I can navigate to next and last page, I cannot go to previous or first page.

**Comment by Vinai** [comment]
Same behavior when you change to a page size of 10?

**Comment by lbuchholz** [comment]
Yes.

**Comment by Vinai** [comment]
I have created 30 sliders, no problem with paging back or forth.
Any errors in the console or the ajax response?

**Comment by Vinai** [comment]
Do you have an answer to my question @lbuchholz? Could you check in the network tab in the devtools if there is some kind of exception or something?

**Comment by lbuchholz** [comment]
@Vinai Sorry for taking such a long time to respond. I just did a complete reinstall of all modules involved et voilà, works. 🙄
Please close, thank you 👍
